test/fail_compilation/fail22084.d

```diff
@@ -0,0 +1,23 @@
+// https://issues.dlang.org/show_bug.cgi?id=22084
+/*
+TEST_OUTPUT
+---
+fail_compilation/fail22084.d(22): Error: cannot pass types that need destruction as variadic arguments
+---
+*/
+import core.stdc.stdarg;
+
+extern(C++) void testVariadic(int a, ...)
+{
+}
+
+struct Destructor
+{
+    ~this() { }
+}
+
+void test()
+{
+    auto a0 = Destructor;
+    testVariadic(1, a0);
+}
```

Comment:
Could also test the scope propagation, e.g. by passing a `scope int* ptr`.

Reply:
I might be a while thinking of a meaningful test that might otherwise fail (Issue 21880, but with variadic arguments).

Unless you mean just passing a scope pointer just for the sake of it. :-)